### PR TITLE
Extract BankId login from SsnSeSection into app-wide service available through context

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Create BankIdContext to handle authentication as app-global concern
- Add global BankIdDialog with authentication UI

## Justify why they are needed

- Ensure BankId auth is always on top of any modal dialogs
- Prepare to handle signing the same way

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1525** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1525" target="_blank">Graphite</a> 
    * **PR #1529** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1529" target="_blank">Graphite</a> 
      * **PR #1558** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1558" target="_blank">Graphite</a> 
        * **PR #1549** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1549" target="_blank">Graphite</a> 
          * **PR #1554** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1554" target="_blank">Graphite</a> 
            * **PR #1565** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1565" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1565?utm_source=stack-comment).

**Review by robinandeer — COMMENTED** [review]
Interesting approach -- I can see the need for reuse here

What I'm concerned about it to create a global concept out of something that we know is a country-specific feature

I would like to discuss whether we should abstract away the country-specific implementations/names if we make this switch. Otherwise it seems like we will sprinkle a lot of "sweden" logic all over the code base which is what we want to avoid

File: apps/store/src/services/bankId/BankIdContext.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+export const BankIdContextProvider = ({ children }: PropsWithChildren) => {
+  const { shopSession } = useShopSession()
+  const [currentOperation, setCurrentOperation] =
+    useState<BankIdContextValue['currentOperation']>(null)
+
+  // TODO: Expose and handle errors
+  const [authenticateShopSession] = useShopSessionAuthenticateMutation()
+
+  const showLoginPrompt = useCallback(() => {
+    console.log('init login')
```
Comment (robinandeer) [inline apps/store/src/services/bankId/BankIdContext.tsx:34]: do you mean to log with datadog?

Comment (alebedev) [reply apps/store/src/services/bankId/BankIdContext.tsx]: Yes, plan to clear this out in the next PR

File: apps/store/src/services/bankId/BankIdContext.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+export const BankIdContextProvider = ({ children }: PropsWithChildren) => {
+  const { shopSession } = useShopSession()
+  const [currentOperation, setCurrentOperation] =
+    useState<BankIdContextValue['currentOperation']>(null)
+
+  // TODO: Expose and handle errors
+  const [authenticateShopSession] = useShopSessionAuthenticateMutation()
+
+  const showLoginPrompt: BankIdContextValue['showLoginPrompt'] = useCallback(({ onCompleted }) => {
+    console.log('show login', { onCompleted })
```
Comment (robinandeer) [inline apps/store/src/services/bankId/BankIdContext.tsx:37]: Datadog?

File: apps/store/src/services/bankId/BankIdContext.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    setCurrentOperation({
```
Comment (robinandeer) [inline apps/store/src/services/bankId/BankIdContext.tsx:38]: Perhaps we should experiment a little with `useReducer` in these more complicated cases -- future improvement

**Review by robinandeer — APPROVED** [review]
A lot to take it but I believe it's moving us in the right direction ⭕

File: apps/store/src/services/bankId/BankIdContext.tsx (hunk 2 above)
Comment (alebedev) [reply apps/store/src/services/bankId/BankIdContext.tsx:37]: Next PR 😄

**Pull request merged**